### PR TITLE
add checksum address

### LINKED ISSUE #1068: [ETH] Convert addresses to checksum address
THORChain / Midgard might not provide checksum addresses, we do need to convert them. That's needed to show icons for ERC20 tokens etc.

### PR DESCRIPTION
**Pull request by sarawut11**
closes #1068

### CODE REVIEW

File: src/renderer/helpers/addressHelper.ts [hunk 1]
```diff
@@ -55,3 +56,5 @@ export const removeAddressPrefix = (address: Address): Address => {
   const prefixIndex = address.indexOf(':') + 1
   return address.substr(prefixIndex > 0 ? prefixIndex : 0)
 }
+
+export const getETHChecksumAddress = (address: Address) => ethers.utils.getAddress(address.toLowerCase())
```
Comment (veado) [inline src/renderer/helpers/addressHelper.ts:60]: Why `address.toLowerCase()` and not just `address`?

Pls add comment:
```ts
/**
* Helper to get ETH address as a checksum address
*/
```

[nitpick] Pls keep function names in common (`getEthXYZ` vs `getETHXYZ` / `getEthTokenAddress` vs. `getETHChecksumAddress`)

**Review by veado — CHANGES_REQUESTED** [review]

File: src/renderer/helpers/addressHelper.test.ts [hunk 2]
```diff
@@ -106,4 +107,16 @@ describe('helpers/addressHelper', () => {
...
+    it('ethereum uppercase address', () => {
+      const result = getEthChecksumAddress('0x0089D53F703F7E0843953D48133F74CE247184C2')
+      expect(result).toEqual(O.some('0x0089d53F703f7E0843953D48133f74cE247184c2'))
+    })
+
+    it('ethereum lowercase address', () => {
+      const result = getEthChecksumAddress('0x0089d53f703f7e0843953d48133f74ce247184c2')
+      expect(result).toEqual(O.some('0x0089d53F703f7E0843953D48133f74cE247184c2'))
+    })
+  })
```
Comment (veado) [inline src/renderer/helpers/addressHelper.test.ts:121]: Pls add a test for failings (`None`)

File: src/renderer/helpers/addressHelper.ts [hunk 3]
```diff
@@ -55,3 +57,12 @@ export const removeAddressPrefix = (address: Address): Address => {
...
   return address.substr(prefixIndex > 0 ? prefixIndex : 0)
 }
+
+/**
+ * Helper to get ETH address as a checksum address
+ * ERC20 addresses start with 0X rather than 0x in Asgardex
+ * Due to that, address.toLowerCase() is needed
+ * If not, ethers getAddress function recognize the address as invalid one
+ */
+export const getEthChecksumAddress = (address: Address) =>
```
Comment (veado) [inline src/renderer/helpers/addressHelper.ts:67]: Pls add a return type: `(address: Address): O.Option<Address>`

File: src/renderer/helpers/assetHelper.ts [hunk 4]
```diff
@@ -112,7 +113,7 @@ export const isChainAsset = (asset: Asset): boolean => eqAsset.equals(asset, get
 /**
  * Get ethereum token address from a given asset
  */
-export const getEthTokenAddress = (asset: Asset): string => {
+export const getEthTokenAddress = (asset: Asset): O.Option<string> => {
```
Comment (veado) [inline src/renderer/helpers/assetHelper.ts:116]: `O.Option<string>` -> `O.Option<Address>`

**Review by veado — CHANGES_REQUESTED** [review]
